### ISSUE TITLE
Trains aren't working correctly in multiplayer.

### ISSUE THREAD

**Issue by novacheri1** [TeamOpenIndustry/ImmersiveRailroading#1068]
**Describe the bug**  
When entering any locomotive or rolling stock, a number of bugs can happen inconsistently, or not at all:
A. You fall through the train, and can move around as you would before entering it, while outside of it, despite the game considering you in the train still. You enter a sort of ghost state, where you can attack mobs yet they don't recognize you, and players will still see you in the train on their side of the game. Locomotive controls aren't displayed, nor responsive during this time. Dismounting like normal sends you back to where you should leave the train.
B: You enter the train, except you don't visually move from car to car, your player only shown sitting stationary in the center of whichever car you're in at the time, as if the car was a minecart. When in the locomotive, attempting to move won't show the train moving, or play any sound affects, yet your player will be sent down the tracks in a jittery manner, skipping from the first place to the next.
EDIT: B seems to happen more frequently if not in the first place after rejoining the server.
C: You can enter the train and move from car to car visually, but you don't seem to be moving from each car functionally. This is shown when you attempt to control a locomotive on the consist, which won't respond to any inputs give. Leaving the train sends you to the first car you entered.

**How to reproduce**  
Steps to reproduce the unwanted behavior:

 1. Start a server with IR installed.
 2. Enter a train.
 3. If effects are not present, you can try entering the train at a later time, or rejoining the server. As stated these bugs are inconsistent.  
 
**Expected behavior**  
Trains should be consistent in control behavior regardless of which car you enter at first. Their sounds and visuals should match whats happening logically. Trains should take you into the standard movement scheme and not allow players to walk around as if they aren't inside of the train, when they functionally are.

**Screenshots**  
Issue A: Walking outside of the train like I never entered it, despite crouch sending me to the exit of the locomotive.
[https://imgur.com/a/0g60Ygn](url)
Issue B: Entering the locomotive, before adding throttle and moving down the tracks.
[https://imgur.com/a/wOlhvVR](url)

**System Information**  
 OS: Arch Linux x86_64
 <br>System Memory: 32GB
 <br>Allocated Memory: 16GB
 <br>CPU: AMD Ryzen 5 3600 (12) @ 3.6GHz
 <br>GPU: NVIDIA GeForce GTX 1050 Ti
 <br>IR-Version: 1.16.5 - 1.9.0
 <br>Forge-Version: 1.16.5 - 36.2.2

**Modlist**  
 Server-side modlist: https://pastebin.com/a7S2hKLX
 Client-side modlist: https://pastebin.com/UJiiw13U
 
**latest.log**
Server Log: https://pastebin.com/7X8nkhEW
Client Log: [latest.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/7081307/latest.log)

**Additional context**  
Using `/immersiverailroading reload` or reloading chunks has no affect on these problems, rejoining the server helps sometimes remove or cause these problems, and restarting the server seems to be an overall fix. These errors always seem to come back in the same session, however. I've tried removing any items that modify movement, yet those had no affect. I was also connecting to the server from the same computer it was hosted on, allocating 12GB.

**Comment by km00700** [comment]
Is this a plugin server? (Mohist, Sponge, Magma, etc)

**Comment by Ghostizockt** [comment]
Ohhh you use Linux, i use linux too and i have the First Bug even in
singeplayer 0-o

Kai Mills [km00700] [DarkRaider] ***@***.***> schrieb am Di.,
31. Aug. 2021, 09:38:

**Comment by Ghostizockt** [comment]
Oops wanted to answer too you(Gmail is annoying)

novacheri ***@***.***> schrieb am Di., 31. Aug. 2021, 06:54:

**Comment by Ghostizockt** [comment]
Also youre using 1.16.5, Same as me, youre using Linux, Same as me, and i
have the First Bug also

novacheri ***@***.***> schrieb am Di., 31. Aug. 2021, 06:54:

**Comment by cam72cam** [comment]
@novacheri have you tried booting up a server with just IR/UMC/TrackAPI and then adding mods a few a time?  There's a good chance it's a mod conflict.

**Comment by cam72cam** [comment]
There's also a mod filling up your client log with garbage and making it unusable:
```
OpenGL API ERROR: 1281 (GL_INVALID_VALUE error generated. neither <first> nor <count> can be negative.)
``

**Comment by novacheri** [comment]
Sorry for the late responses, couldn't get a chance to tackle these problems until an hour ago.

> Is this a plugin server? (Mohist, Sponge, Magma, etc)

We're running a stock Forge server. Client and server are at equal versions.

> @novacheri have you tried booting up a server with just IR/UMC/TrackAPI and then adding mods a few a time?
There's a good chance it's a mod conflict.

I haven't done this yet, but I'll get to this either tomorrow or tonight since I already spent a good chunk of time trying to remove the garbage in the log file. Speaking of which;

> There's also a mod filling up your client log with garbage and making it unusable:
> 
> ```
> OpenGL API ERROR: 1281 (GL_INVALID_VALUE error generated. neither <first> nor <count> can be negative.)
> ``
> ```

I figured out this is an Optifine related issue with render regions, I've turned that off now. Here's a new log file, if it gives anymore insight until I manage to go through all the mods. Error A was happening for the most part on the first join, Error B started happening consistently (except for one A instance) on the second join.
Standard Log: https://pastebin.com/LWCsGEsv
Debug Log (In case its useful): [debug.log](https://github.com/TeamOpenIndustry/ImmersiveRailroading/files/7087680/debug.log)
